src/App.tsx

```diff
@@ -17,18 +63,49 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setFilterBy={setFilterBy}
+                filterBy={filterBy}
+                setQuery={(queryFromInput) => {
+                  setQuery(queryFromInput);
+                }}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading
+                ? <Loader />
+                : (
+                  <TodoList
+                    todos={filteredTodos}
+                    selectedTodoId={selectedTodo}
+                    selectedTodo={(id) => {
+                      setSelectedTodo(id);
+                    }}
+                    selectedUserId={(userId) => {
```

Comment:
Pay attention on names in same places all over the task.
```suggestion
                    setSelectedUserId={(userId) => {
```